framework/src/bcs/NodalBC.C

```diff
@@ -43,6 +77,13 @@ NodalBC::computeResidual(NumericVector<Number> & residual)
     dof_id_type & dof_idx = _var.nodalDofIndex();
     _qp = 0;
     residual.set(dof_idx, computeQpResidual());
+
+    if (_has_save_in)
+    {
+      Threads::spin_mutex::scoped_lock lock(Threads::spin_mtx);
+      for (unsigned int i=0; i<_save_in.size(); i++)
+        _save_in[i]->sys().solution().set(dof_idx, 0);
```

Comment:
I see. That is something I was thinking might not be right.

Because we are setting the value at nodes and the residual contribution of these nodes should be just zero.